packages/util/src/bytes.ts

```diff
@@ -24,20 +24,20 @@ export const bytesToUnprefixedHex = _bytesToUnprefixedHex
  * @returns {Uint8Array} The converted bytes
  * @throws If the input is not a valid 0x-prefixed hex string
  */
-export const hexToBytes = (hex: string) => {
+export const hexToBytes = (hex: PrefixedHexString): Uint8Array => {
   if (!hex.startsWith('0x')) throw EthereumJSErrorWithoutCode('input string must be 0x prefixed')
   return nobleH2B(padToEven(stripHexPrefix(hex)))
 }
 
-export const unprefixedHexToBytes = (hex: string) => {
+export const unprefixedHexToBytes = (hex: string): Uint8Array => {
   if (hex.startsWith('0x')) throw EthereumJSErrorWithoutCode('input string cannot be 0x prefixed')
   return nobleH2B(padToEven(hex))
 }
 
 export const bytesToHex = (bytes: Uint8Array): PrefixedHexString => {
   if (bytes === undefined || bytes.length === 0) return '0x'
```

Comment:
Can we remove this `bytes === undefined` here? And can we also add a simple test for the `undefined` case?

Reply:
I think that `undefined` should throw, and not return empty bytes :thinking:

Reply:
Agreed, have some pending work that I'll push soon

Reply:
You've seen it already, but for reference this will be addressed in a separate PR and I created an issue for it here #3998